transport/endpoint.go

```diff
@@ -0,0 +1,108 @@
+package transport
+
+import (
+	"context"
+	"net/http"
+	"github.com/playnet-public/apiexperiments/problems"
+	"github.com/playnet-public/apiexperiments/encode"
+	"go.uber.org/zap"
+)
+
+//HandlerFunc of the main service functions
+type HandlerFunc func(ctx context.Context, r *http.Request) (interface{}, error)
+
+//MiddlewareFunc of middlewares surrounding the main service
+type MiddlewareFunc func(ctx context.Context) error
```

Comment:
where is the sense for a middleware not even interacting with the request, data or endpoint?
Can you provide a sample on how this should be used/work?

Reply:
Changed to `func (context.Context, r *http.Request) error`